packages/protocol/test/governance/network/epochrewards.ts

```diff
@@ -657,6 +657,221 @@ contract('EpochRewards', (accounts: string[]) => {
+            )
+          )
+        )
+        const actual = fromFixed((await epochRewards.getTargetVotingYieldParameters())[0])
+        // Assert equal to 9 decimal places due to fixidity imprecision.
+        assert.equal(expected.dp(9).toFixed(), actual.dp(9).toFixed())
+      })
+    })
+
+    // Maximum target voting yield is enforced
+    describe('When target voting yield is increased by adjustment factor', () => {
+      beforeEach(async () => {
+        const votingFraction = 0.1
+        const totalVotes = floatingSupply.times(votingFraction).integerValue(BigNumber.ROUND_FLOOR)
+        await mockElection.setTotalVotes(totalVotes)
+        // naive time travel: mining takes to long, just repeatedly update target voting yield. One call is one epoch travelled
+        for (let i = 0; i < 600; i++) {
+          await epochRewards.updateTargetVotingYield()
+        }
+      })
+
+      it('maximum target voting yield should be enforced', async () => {
+        const expected = fromFixed(targetVotingYieldParams.max)
+        const actual = fromFixed((await epochRewards.getTargetVotingYieldParameters())[0])
+        // Assert equal to 9 decimal places due to fixidity imprecision.
+        assert.equal(expected.dp(9).toFixed(), actual.dp(9).toFixed())
+      })
+    })
+
+    // Minimum target voting yield is enforced
+    describe('When target voting yield is decreased by adjustment factor', () => {
+      beforeEach(async () => {
+        const votingFraction = 0.89
+        const totalVotes = floatingSupply.times(votingFraction).integerValue(BigNumber.ROUND_FLOOR)
+        await mockElection.setTotalVotes(totalVotes)
+        // naive time travel: mining takes to long, just repeatedly update target voting yield. One call is one epoch travelled
+        for (let i = 0; i < 800; i++) {
+          await epochRewards.updateTargetVotingYield()
+        }
+      })
+
+      it('minimum target voting yield of 0 should be enforced', async () => {
+        const expected = fromFixed(toFixed(0))
+        const actual = fromFixed((await epochRewards.getTargetVotingYieldParameters())[0])
+        // Assert equal to 9 decimal places due to fixidity imprecision.
+        assert.equal(expected.dp(9).toFixed(), actual.dp(9).toFixed())
+      })
+    })
+
+    // test several epochs of the same adjustment in a row
+    describe('When voting fraction remains below target 5 epochs in a row', () => {
+      beforeEach(async () => {
+        const votingFraction = 0.3
+        const totalVotes = floatingSupply.times(votingFraction).integerValue(BigNumber.ROUND_FLOOR)
+        await mockElection.setTotalVotes(totalVotes)
+        // naive time travel: mining blocks takes to long, just repeatedly update target voting yield. One call is one epoch travelled
+        for (let i = 0; i < 5; i++) {
+          await epochRewards.updateTargetVotingYield()
+        }
+      })
+
+      it('target voting yield should be increased 5 times as expected', async () => {
```

Comment:
How about adding another test that shows how a certain adjustmentfactor is supposed to effect the yield economically of 365 days? E.g. using the new value proposed in your CGP and checking for whether the 1 year change is as expected in the scenario used to derive this value.

Reply:
Do you imagine them in the 'per epoch' or 'annualized' world? I would recommend against adding tests for annualized yields here as everything in the code happens in the 'per epoch' world. I think the docs would be a good place to explain the annualization and epoch / yearly yield difference. 

I added two test cases below to test the 365 day change in the 'per epoch' world, that's a good idea

Reply:
Yeah definitely in epochs, was a bit imprecise here. The test cases look great! But would probably rephrase some of the comments to "365 epochs (approximately 365 days, i.e. one year)".

Reply:
Good point, changed that